x-pack/solutions/security/plugins/security_solution/public/siem_migrations/rules/components/rules_table/start_rule_migration_modal/start_rule_migration_modal.tsx

```diff
@@ -0,0 +1,212 @@
+          dropdownDisplay: (
+            <>
+              <strong>{connector.name}</strong>
+              <EuiText size="s" color="subdued">
+                <p>{connectorDescription}</p>
+              </EuiText>
+            </>
+          ),
+          'data-test-subj': `${DATA_TEST_SUBJ_PREFIX}-ConnectorOption`,
+        };
+      });
+    }, [actionTypeRegistry, availableConnectors]);
+
+    const getSecuritySolutionLinkProps = useGetSecuritySolutionLinkProps();
+
+    const { onClick: onClickSetupAIConnector, href: setupAIConnectorLink } =
+      getSecuritySolutionLinkProps({
+        deepLinkId: SecurityPageName.landing,
+        path: `${OnboardingTopicId.siemMigrations}#${OnboardingCardId.siemMigrationsAiConnectors}`,
+      });
+
+    const onStartMigrationWithSettings: React.FormEventHandler = useCallback(
+      (e) => {
+        e.preventDefault();
+        if (!selectedConnectorId) {
+          return;
+        }
+        startMigrationWithSettings({
+          connectorId: selectedConnectorId,
+          skipPrebuiltRulesMatching: !enablePrebuiltRulesMatching,
+        });
+        closeModal();
+      },
+      [startMigrationWithSettings, selectedConnectorId, enablePrebuiltRulesMatching, closeModal]
+    );
+
+    return (
+      <EuiOutsideClickDetector onOutsideClick={closeModal}>
+        <EuiModal onClose={closeModal} data-test-subj={DATA_TEST_SUBJ_PREFIX}>
+          <EuiModalHeader>
+            <EuiModalHeaderTitle
+              data-test-subj={`${DATA_TEST_SUBJ_PREFIX}-Title`}
+              id={modalTitleId}
+            >
+              {i18n.REPROCESS_RULES_DIALOG_TITLE(numberOfRules)}
+            </EuiModalHeaderTitle>
+          </EuiModalHeader>
+          <EuiModalBody>
+            <EuiText>
+              <p>{i18n.REPROCESS_RULES_DIALOG_DESCRIPTION}</p>
+            </EuiText>
+            <EuiSpacer size="m" />
+            <EuiForm
+              component="form"
+              onSubmit={onStartMigrationWithSettings}
+              data-test-subj={`${DATA_TEST_SUBJ_PREFIX}-Form`}
+            >
+              <EuiFormRow
+                label={i18n.REPROCESS_RULES_DIALOG_AI_CONNECTOR_LABEL}
+                helpText={
+                  <FormattedMessage
+                    id="xpack.securitySolution.siemMigrations.reprocessFailedRulesDialog.connectorHelpText"
+                    defaultMessage={'To setup other LLM connectors, visit {link}.'}
+                    values={{
+                      link: (
+                        /* eslint-disable-next-line @elastic/eui/href-or-on-click */
+                        <EuiLink href={setupAIConnectorLink} onClick={onClickSetupAIConnector}>
+                          {i18n.REPROCESS_RULES_DIALOG_SETUP_NEW_AI_CONNECTOR_HELP_TEXT}
+                        </EuiLink>
+                      ),
+                    }}
+                  />
+                }
+                isInvalid={!selectedConnectorId}
+              >
+                <EuiSuperSelect
```

Comment:
We have a generic [`ConnectorSelector`](https://github.com/elastic/kibana/blob/main/x-pack/solutions/security/packages/connectors/src/connector_selector.tsx) component that we can reuse. Have you tried using it?

Reply:
Okay this is done in e4776b0bf00f.

I know you were talking about a new component that encapsulates `ConnectorSelector` and `useAIConnectors` . 
Let's do that when we refactor these mutiple `ConnectorSelector` and `useConnectors`

Reply:
In previous commit, i added an extra onClick handler, which i removed here: https://github.com/elastic/kibana/pull/222542/commits/03aa99069942b98f63f55b697436b103407b0b2f